sound/soc/sof/intel/hda-ipc.c

```diff
@@ -89,8 +89,19 @@ int hda_dsp_ipc_get_reply(struct snd_sof_dev *sdev,
 	int ret = 0;
 	u32 size;
 
-	/* get IPC reply from DSP in the mailbox */
-	sof_mailbox_read(sdev, sdev->host_box.offset, &reply, sizeof(reply));
+	if (msg->header == (SOF_IPC_GLB_PM_MSG | SOF_IPC_PM_CTX_SAVE)) {
```

Comment:
I thought we discussed that the msg->header field would no longer be used, see e.g. https://github.com/thesofproject/linux/pull/586
May be that's a good counter-example that it should? Or can you use the body of the msg to find thhe command information

Reply:
@plbossart Yes, I can use the body to find the command information as ```struct sof_ipc_cmd_hdr``` is always the first element of ```msg_data```.  However, I think check ```msg->header``` doesn't against #586.  We don't write ``msg->header`` to SHIM register in #586, but it doesn't touch ``msg->header``. In other word, we can still use ``msg->header`` after #586 is merged.

Reply:
no the idea was that the msg->header would be removed if it serves no purpose.

Reply:
@plbossart updated. use the body of the msg to check if it is SOF_IPC_PM_CTX_SAVE